### PR TITLE
GumGum: adds slot param

### PR DESCRIPTION
**Pull request by susyt**
allows publishers to optionally send in a slot param 

update to prebid doc to come

### CODE REVIEW

**Comment by lbenmore** [comment]
PR for updates to prebid docs: https://github.com/prebid/prebid-server/pull/1949#issue-704363088

File: adapters/gumgum/gumgum.go [hunk 1]
```diff
@@ -146,6 +146,34 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
 		format := bannerCopy.Format[0]
 		bannerCopy.W = &(format.W)
 		bannerCopy.H = &(format.H)
+
+		if gumgumExt.Slot != 0 {
```
Comment (guscarreon) [inline adapters/gumgum/gumgum.go:150]: In order to keep the `preprocess(imp *openrtb2.Imp)` function short, can the newly added functionality be grouped into a function of its own? Maybe a `getBiggerFormat(formatList []openrtb2.Format, slot float64) openrtb_ext.ExtImpGumGumBanner` could help.
```
diff --git a/adapters/gumgum/gumgum.go b/adapters/gumgum/gumgum.go
index 4d375af3..9688854a 100644
--- a/adapters/gumgum/gumgum.go
+++ b/adapters/gumgum/gumgum.go
@@ -148,26 +148,8 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
                bannerCopy.H = &(format.H)

                if gumgumExt.Slot != 0 {
-                       maxw := int64(0)
-                       maxh := int64(0)
-                       greatestVal := int64(0)
-                       for _, size := range bannerCopy.Format {
-                               var biggerSide int64
-                               if size.W > size.H {
-                                       biggerSide = size.W
-                               } else {
-                                       biggerSide = size.H
-                               }
-
-                               if biggerSide > greatestVal || (biggerSide == greatestVal && size.W >= maxw && size.H >= maxh) {
-                                       greatestVal = biggerSide
-                                       maxh = size.H
-                                       maxw = size.W
-                               }
-                       }
-
+                       bannerExt := getBiggerFormat(bannerCopy.Format, gumgumExt.Slot)
                        var err error
-                       bannerExt := openrtb_ext.ExtImpGumGumBanner{Si: gumgumExt.Slot, MaxW: float64(maxw), MaxH: float64(maxh)}
                        bannerCopy.Ext, err = json.Marshal(&bannerExt)
                        if err != nil {
                                return nil, err
@@ -197,6 +179,28 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
        return &gumgumExt, nil
 }

+func getBiggerFormat(formatList []openrtb2.Format, slot float64) openrtb_ext.ExtImpGumGumBanner {
+       maxw := int64(0)
+       maxh := int64(0)
+       greatestVal := int64(0)
+       for _, size := range formatList {
+               var biggerSide int64
+               if size.W > size.H {
+                       biggerSide = size.W
+               } else {
+                       biggerSide = size.H
+               }
+
+               if biggerSide > greatestVal || (biggerSide == greatestVal && size.W >= maxw && size.H >= maxh) {
+                       greatestVal = biggerSide
+                       maxh = size.H
+                       maxw = size.W
+               }
+       }
+
+       return openrtb_ext.ExtImpGumGumBanner{Si: slot, MaxW: float64(maxw), MaxH: float64(maxh)}
+}
```

**Review by guscarreon — COMMENTED** [review]
Left a nitpick. I'm totally ok if you don't agree, I'm ready to approve either way.

File: adapters/gumgum/gumgum.go (hunk 1 above)
Comment (susyt) [reply adapters/gumgum/gumgum.go:150]: thanks @guscarreon ! I've updated the PR to move it into a `getBiggerFormat` func - please let me know if there are any other changes/suggestions you might have

Comment (guscarreon) [reply adapters/gumgum/gumgum.go:150]: Thank you @susyt

**Review by guscarreon — APPROVED** [review]
LGTM

**Review by AlexBVolcy — APPROVED** [review]
I agreed with Gus' comment about creating that `getBiggerFormat` function, but i think that looks great! Apart from that, LGTM!

**Pull request merged**